lightning/src/chain/chainmonitor.rs

```diff
@@ -128,6 +186,37 @@ where C::Target: chain::Filter,
 		}
 	}
 
+	/// Dispatches to per-channel monitors, which are responsible for updating their on-chain view
+	/// of a channel based on transactions unconfirmed as a result of a chain reorganization. See
+	/// [`ChannelMonitor::transaction_unconfirmed`] for details.
+	///
+	/// Used instead of [`block_disconnected`] by clients that are notified of transactions rather
+	/// than blocks. May be called before or after [`update_best_block`] for transactions in the
+	/// corresponding block. See [`update_best_block`] for further calling expectations. 
+	///
+	/// [`block_disconnected`]: Self::block_disconnected
+	/// [`update_best_block`]: Self::update_best_block
+	pub fn transaction_unconfirmed(&self, txid: &Txid) {
+		let monitors = self.monitors.read().unwrap();
+		for monitor in monitors.values() {
+			monitor.transaction_unconfirmed(txid, &*self.broadcaster, &*self.fee_estimator, &*self.logger);
+		}
+	}
+
+	/// Returns the set of txids that should be monitored for confirmation on chain or for
+	/// re-organization out of the chain.
```

Comment:
I don't think "confirmation on chain" is correct here - we only care about these transactions for reorganization out of the chain (and probably it could be named better in the next PR).

Reply:
We should ultimately end up with one set of/the same docs, but for the sake of having two starting points, I took a crack at from-scratch docs at https://github.com/rust-bitcoin/rust-lightning/pull/853/commits/4d11f31395f1c7ae70f2f73e36d823a8918e5fd9